Scripts/public-api-diff/Sources/Helpers/ProjectHelper.swift

```diff
@@ -0,0 +1,169 @@
+        
+        let sourceDirectoryPath: String
+        
+        switch source {
+        case let .local(path):
+            sourceDirectoryPath = path
+            
+        case let .remote(branchOrTag, repository):
+            sourceDirectoryPath = retrieveRemoteProject(branchOrTag: branchOrTag, repository: repository)
+        }
+
+        let sourceWorkingDirectoryPath = workingDirectoryPath.appending("/\(UUID().uuidString)")
+        
+        try Self.setupIndividualWorkingDirectory(
+            at: sourceWorkingDirectoryPath,
+            sourceDirectoryPath: sourceDirectoryPath,
+            fileHandler: fileHandler,
+            shell: shell
+        )
+        
+        let packagePath = PackageFileHelper.packagePath(for: sourceWorkingDirectoryPath)
+        
+        try Self.buildProject(
+            projectDirectoryPath: sourceWorkingDirectoryPath,
+            packageFileHelper: PackageFileHelper(packagePath: packagePath, fileHandler: fileHandler),
+            xcodeTools: XcodeTools(shell: shell)
+        )
+        
+        switch source {
+        case .local:
+            break
+        case .remote:
+            // Clean up the cloned repo
+            try? fileHandler.removeItem(atPath: sourceDirectoryPath)
+        }
+        
+        return sourceWorkingDirectoryPath
+    }
+    
+    static func setupIndividualWorkingDirectory(
+        at destinationDirectoryPath: String,
+        sourceDirectoryPath: String,
+        fileHandler: FileHandling,
+        shell: ShellHandling
+    ) throws {
+        
+        try? fileHandler.removeItem(atPath: destinationDirectoryPath)
+        try fileHandler.createDirectory(atPath: destinationDirectoryPath)
+        
+        let fileNameIgnoreList: Set<String> = [
+            ".build",
+            ".git",
+            ".github",
+            ".gitmodules",
+            ".codebeatignore",
+            ".swiftformat",
+            ".swiftpm",
+            ".DS_Store",
+            "Tests",
+            "Cartfile"
+        ]
+        
+        let fileExtensionIgnoreList: Set<String> = [
+            ".yaml",
+            ".yml",
+            ".png",
+            ".docc",
+            ".xcodeproj",
+            ".xcworkspace",
+            ".md",
+            ".resolved",
+            ".podspec"
+        ]
+        
+        try fileHandler.contentsOfDirectory(atPath: sourceDirectoryPath).forEach { fileName in
+            if fileExtensionIgnoreList.contains(where: { fileName.hasSuffix($0) }) { return }
+            if fileNameIgnoreList.contains(where: { fileName == $0 }) { return }
+            
+            let sourceFilePath = sourceDirectoryPath.appending("/\(fileName)")
+            let destinationFilePath = destinationDirectoryPath.appending("/\(fileName)")
+            
+            // Using shell here as it's faster than the FileManager
+            shell.execute("cp -a \(sourceFilePath) \(destinationFilePath)")
```

Comment:
Do we sanitize user input anywhere? I quickly scanned changes and didn't find it right away.

Reply:
No there is no explicit sanitization going on here but because of previous calls we know that the paths are valid file paths. Might want to revisit this decision though